src/humanloop/evals/run.py

```diff
@@ -366,9 +380,125 @@ class _LocalEvaluator:
     function: Callable
 
 
-def _callable_is_hl_utility(file: File) -> bool:
+EvaluatedFile = Union[PromptResponse, FlowResponse, ToolResponse, EvaluatorResponse, AgentResponse]
+HumanloopSubclient = Union[PromptsClient, FlowsClient, ToolsClient, EvaluatorsClient, AgentsClient]
+
+
+def _get_subclient(client: "BaseHumanloop", file_config: FileEvalConfig) -> HumanloopSubclient:
+    """Get the appropriate subclient based on file type."""
+    type_ = file_config.get("type")
+    if type_ == "prompt":
+        return client.prompts
+    elif type_ == "flow":
+        return client.flows
+    elif type_ == "tool":
+        return client.tools
+    elif type_ == "evaluator":
+        return client.evaluators
+    elif type_ == "agent":
+        return client.agents
+    else:
+        raise HumanloopRuntimeError(f"Unsupported File type: {type_}")
+
+
+def _safe_get_default_file_version(client: "BaseHumanloop", file_config: FileEvalConfig) -> EvaluatedFile:
+    """Get default version of a File from online workspace.
+
+    Uses either the File path or id from the config.
+
+    Raise error if the File is not of the expected type, or if the user has provided both a path and an id.
+    """
+    path = file_config.get("path")
+    type = file_config.get("type")
+    file_id = file_config.get("id")
+
+    if path is None and file_id is None:
+        raise HumanloopRuntimeError("You must provide a path or id in your `file`.")
+
+    if path is not None:
+        hl_file = client.files.retrieve_by_path(path=path)
+        if hl_file.type != type:
+            raise HumanloopRuntimeError(
+                f"File in Humanloop workspace at {path} is not of type {type}, but {hl_file.type}."
+            )
+        # cast is safe, we can only fetch Files allowed by FileType
+        return typing.cast(EvaluatedFile, hl_file)
+    elif file_id is not None:
+        subclient = _get_subclient(client=client, file_config=file_config)
+        return subclient.get(id=file_id)
+    else:
+        raise HumanloopRuntimeError("You must provide a path or id in your `file` config.")
```

Comment:
Maybe "You must provide either a path or id in your `file` config". Or something to indicate it's an XOR

Reply:
ack